src/Core/Services/Implementations/UserService.cs

```diff
@@ -364,6 +370,17 @@ public async Task<CredentialCreateOptions> StartWebAuthnRegistrationAsync(User u
             provider.MetaData = new Dictionary<string, object>();
         }
 
+        // Boundary validation to provide a better UX. There is also second-level enforcement at persistence time.
+        var maximumAllowedCredentialCount = await _hasPremiumAccessQuery.HasPremiumAccessAsync(user.Id)
+            ? _globalSettings.WebAuthn.PremiumMaximumAllowedCredentials
+            : _globalSettings.WebAuthn.NonPremiumMaximumAllowedCredentials;
+        // Allow 'pending' requests -- including the incoming.
+        if (provider.MetaData.Count(metadata => !metadata.Key.Equals("pending", StringComparison.OrdinalIgnoreCase)) >=
+            maximumAllowedCredentialCount)
```

Comment:
❓ : I was reviewing the clients PR and I wanted to understand the early validation put in place upon read start so I came here to see this. I'm not sure I understand the comment above this line. The code is checking if non pending keys are greater than or equal to the max allowed credential limit. Thus, it's technically excluding pending requests against the limit. 

Could we rewrite this to say something like: "// Exclude 'pending' requests from count (including the incoming pending request)."